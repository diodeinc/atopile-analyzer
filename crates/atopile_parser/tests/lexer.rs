use atopile_parser::lexer::lex;
use std::fs;

macro_rules! create_lexer_test {
    ($name:ident) => {
        #[test]
        fn $name() {
            let input = fs::read_to_string(concat!(
                "tests/resources/corpus/",
                stringify!($name),
                ".ato"
            ))
            .unwrap();
            let normalized_input = input.replace("\r\n", "\n");
            let output = lex(&normalized_input);
            insta::assert_debug_snapshot!(output);
        }
    };
}

create_lexer_test!(vdivs);
create_lexer_test!(resistors);
create_lexer_test!(transistors);
