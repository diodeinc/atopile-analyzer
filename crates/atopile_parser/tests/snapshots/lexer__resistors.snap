---
source: crates/atopile_parser/tests/lexer.rs
expression: output
---
(
    [
        Spanned(
            Import,
            0..6,
        ),
        Spanned(
            Name(
                "Power",
            ),
            7..12,
        ),
        Spanned(
            From,
            13..17,
        ),
        Spanned(
            String(
                "interfaces.ato",
            ),
            18..34,
        ),
        Spanned(
            Newline,
            34..35,
        ),
        Spanned(
            Import,
            35..41,
        ),
        Spanned(
            Name(
                "I2C",
            ),
            42..45,
        ),
        Spanned(
            From,
            46..50,
        ),
        Spanned(
            String(
                "interfaces.ato",
            ),
            51..67,
        ),
        Spanned(
            Newline,
            67..68,
        ),
        Spanned(
            Newline,
            68..69,
        ),
        Spanned(
            Component,
            69..78,
        ),
        Spanned(
            Name(
                "Resistor",
            ),
            79..87,
        ),
        Spanned(
            Colon,
            87..88,
        ),
        Spanned(
            Newline,
            88..89,
        ),
        Spanned(
            Indent,
            89..89,
        ),
        Spanned(
            Signal,
            93..99,
        ),
        Spanned(
            Name(
                "p1",
            ),
            100..102,
        ),
        Spanned(
            Tilde,
            103..104,
        ),
        Spanned(
            Pin,
            105..108,
        ),
        Spanned(
            Number(
                "1",
            ),
            109..110,
        ),
        Spanned(
            Newline,
            110..111,
        ),
        Spanned(
            Signal,
            115..121,
        ),
        Spanned(
            Name(
                "p2",
            ),
            122..124,
        ),
        Spanned(
            Tilde,
            125..126,
        ),
        Spanned(
            Pin,
            127..130,
        ),
        Spanned(
            Number(
                "2",
            ),
            131..132,
        ),
        Spanned(
            Newline,
            132..133,
        ),
        Spanned(
            Name(
                "designator_prefix",
            ),
            137..154,
        ),
        Spanned(
            Equals,
            155..156,
        ),
        Spanned(
            String(
                "R",
            ),
            157..160,
        ),
        Spanned(
            Newline,
            160..161,
        ),
        Spanned(
            Name(
                "mpn",
            ),
            165..168,
        ),
        Spanned(
            Equals,
            169..170,
        ),
        Spanned(
            String(
                "generic_resistor",
            ),
            171..189,
        ),
        Spanned(
            Newline,
            189..190,
        ),
        Spanned(
            Name(
                "type",
            ),
            194..198,
        ),
        Spanned(
            Equals,
            199..200,
        ),
        Spanned(
            String(
                "resistor",
            ),
            201..211,
        ),
        Spanned(
            Newline,
            211..212,
        ),
        Spanned(
            Name(
                "value",
            ),
            216..221,
        ),
        Spanned(
            Colon,
            221..222,
        ),
        Spanned(
            Name(
                "resistance",
            ),
            223..233,
        ),
        Spanned(
            Newline,
            233..234,
        ),
        Spanned(
            Newline,
            234..235,
        ),
        Spanned(
            Comment(
                " Equations",
            ),
            239..250,
        ),
        Spanned(
            Newline,
            250..251,
        ),
        Spanned(
            Comment(
                " R = V/I",
            ),
            255..264,
        ),
        Spanned(
            Newline,
            264..265,
        ),
        Spanned(
            Comment(
                " R = V^2/P",
            ),
            269..280,
        ),
        Spanned(
            Newline,
            280..281,
        ),
        Spanned(
            Newline,
            281..282,
        ),
        Spanned(
            Dedent,
            282..282,
        ),
        Spanned(
            Module,
            282..288,
        ),
        Spanned(
            Name(
                "I2CPullup",
            ),
            289..298,
        ),
        Spanned(
            Colon,
            298..299,
        ),
        Spanned(
            Newline,
            299..300,
        ),
        Spanned(
            Indent,
            300..300,
        ),
        Spanned(
            Name(
                "r_sda",
            ),
            304..309,
        ),
        Spanned(
            Equals,
            310..311,
        ),
        Spanned(
            New,
            312..315,
        ),
        Spanned(
            Name(
                "Resistor",
            ),
            316..324,
        ),
        Spanned(
            Newline,
            324..325,
        ),
        Spanned(
            Name(
                "r_scl",
            ),
            329..334,
        ),
        Spanned(
            Equals,
            335..336,
        ),
        Spanned(
            New,
            337..340,
        ),
        Spanned(
            Name(
                "Resistor",
            ),
            341..349,
        ),
        Spanned(
            Newline,
            349..350,
        ),
        Spanned(
            Name(
                "power",
            ),
            354..359,
        ),
        Spanned(
            Equals,
            360..361,
        ),
        Spanned(
            New,
            362..365,
        ),
        Spanned(
            Name(
                "Power",
            ),
            366..371,
        ),
        Spanned(
            Newline,
            371..372,
        ),
        Spanned(
            Name(
                "i2c",
            ),
            376..379,
        ),
        Spanned(
            Equals,
            380..381,
        ),
        Spanned(
            New,
            382..385,
        ),
        Spanned(
            Name(
                "I2C",
            ),
            386..389,
        ),
        Spanned(
            Newline,
            389..390,
        ),
        Spanned(
            Newline,
            390..391,
        ),
        Spanned(
            Name(
                "r_sda",
            ),
            395..400,
        ),
        Spanned(
            Dot,
            400..401,
        ),
        Spanned(
            Name(
                "value",
            ),
            401..406,
        ),
        Spanned(
            Equals,
            407..408,
        ),
        Spanned(
            Number(
                "10",
            ),
            409..411,
        ),
        Spanned(
            Name(
                "kohm",
            ),
            411..415,
        ),
        Spanned(
            PlusOrMinus,
            416..419,
        ),
        Spanned(
            Number(
                "20",
            ),
            420..422,
        ),
        Spanned(
            Percent,
            422..423,
        ),
        Spanned(
            Newline,
            423..424,
        ),
        Spanned(
            Name(
                "r_scl",
            ),
            428..433,
        ),
        Spanned(
            Dot,
            433..434,
        ),
        Spanned(
            Name(
                "value",
            ),
            434..439,
        ),
        Spanned(
            Equals,
            440..441,
        ),
        Spanned(
            Number(
                "10",
            ),
            442..444,
        ),
        Spanned(
            Name(
                "kohm",
            ),
            444..448,
        ),
        Spanned(
            PlusOrMinus,
            449..452,
        ),
        Spanned(
            Number(
                "20",
            ),
            453..455,
        ),
        Spanned(
            Percent,
            455..456,
        ),
        Spanned(
            Newline,
            456..457,
        ),
        Spanned(
            Newline,
            457..458,
        ),
        Spanned(
            Name(
                "r_sda",
            ),
            462..467,
        ),
        Spanned(
            Dot,
            467..468,
        ),
        Spanned(
            Name(
                "package",
            ),
            468..475,
        ),
        Spanned(
            Equals,
            476..477,
        ),
        Spanned(
            String(
                "0402",
            ),
            478..484,
        ),
        Spanned(
            Newline,
            484..485,
        ),
        Spanned(
            Name(
                "r_scl",
            ),
            489..494,
        ),
        Spanned(
            Dot,
            494..495,
        ),
        Spanned(
            Name(
                "package",
            ),
            495..502,
        ),
        Spanned(
            Equals,
            503..504,
        ),
        Spanned(
            String(
                "0402",
            ),
            505..511,
        ),
        Spanned(
            Newline,
            511..512,
        ),
        Spanned(
            Newline,
            512..513,
        ),
        Spanned(
            Name(
                "i2c",
            ),
            517..520,
        ),
        Spanned(
            Dot,
            520..521,
        ),
        Spanned(
            Name(
                "sda",
            ),
            521..524,
        ),
        Spanned(
            Tilde,
            525..526,
        ),
        Spanned(
            Name(
                "r_sda",
            ),
            527..532,
        ),
        Spanned(
            Dot,
            532..533,
        ),
        Spanned(
            Number(
                "1",
            ),
            533..534,
        ),
        Spanned(
            Semicolon,
            534..535,
        ),
        Spanned(
            Name(
                "r_sda",
            ),
            536..541,
        ),
        Spanned(
            Dot,
            541..542,
        ),
        Spanned(
            Number(
                "2",
            ),
            542..543,
        ),
        Spanned(
            Tilde,
            544..545,
        ),
        Spanned(
            Name(
                "power",
            ),
            546..551,
        ),
        Spanned(
            Dot,
            551..552,
        ),
        Spanned(
            Name(
                "vcc",
            ),
            552..555,
        ),
        Spanned(
            Newline,
            555..556,
        ),
        Spanned(
            Name(
                "i2c",
            ),
            560..563,
        ),
        Spanned(
            Dot,
            563..564,
        ),
        Spanned(
            Name(
                "scl",
            ),
            564..567,
        ),
        Spanned(
            Tilde,
            568..569,
        ),
        Spanned(
            Name(
                "r_scl",
            ),
            570..575,
        ),
        Spanned(
            Dot,
            575..576,
        ),
        Spanned(
            Number(
                "1",
            ),
            576..577,
        ),
        Spanned(
            Semicolon,
            577..578,
        ),
        Spanned(
            Name(
                "r_scl",
            ),
            579..584,
        ),
        Spanned(
            Dot,
            584..585,
        ),
        Spanned(
            Number(
                "2",
            ),
            585..586,
        ),
        Spanned(
            Tilde,
            587..588,
        ),
        Spanned(
            Name(
                "power",
            ),
            589..594,
        ),
        Spanned(
            Dot,
            594..595,
        ),
        Spanned(
            Name(
                "vcc",
            ),
            595..598,
        ),
        Spanned(
            Dedent,
            598..598,
        ),
    ],
    [],
)
